Source/Dronecode-SDK-Swift/Dronecode-SDK-Swift/Camera.swift

```diff
@@ -0,0 +1,492 @@
+        return Observable.create { observer in
+            let cameraModeRequest = Dronecore_Rpc_Camera_SubscribeModeRequest()
+            
+            do {
+                let call = try self.service.subscribemode(cameraModeRequest, completion: nil)
+                while let response = try? call.receive() {
+                    let cameraMode = CameraMode.translateFromRPC(response.cameraMode)
+                    observer.onNext(cameraMode)
+                }
+            } catch {
+                observer.onError("Failed to subscribe to camera mode stream. \(error)")
+            }
+            return Disposables.create()
+        }
+    }
+    
+    public func setVideoStreamSettings(settings: VideoStreamSettings) -> Completable {
+        return Completable.create { completable in
+            var setVideoStreamSettingsRequest = Dronecore_Rpc_Camera_SetVideoStreamSettingsRequest()
+            setVideoStreamSettingsRequest.videoStreamSettings = settings.rpcVideoStreamSettings
+
+            do {
+                let _ = try self.service.setvideostreamsettings(setVideoStreamSettingsRequest)
+                completable(.completed)
+            } catch {
+                completable(.error(error))
+            }
+            
+            return Disposables.create()
+        }
+    }
+    
+    private func createVideoStreamInfoObservable() -> Observable<VideoStreamInfo> {
+        return Observable.create { observer in
+            let videoStreamInfoRequest = Dronecore_Rpc_Camera_SubscribeVideoStreamInfoRequest()
+            
+            do {
+                let call = try self.service.subscribevideostreaminfo(videoStreamInfoRequest, completion: nil)
+                while let response = try? call.receive() {
+                    let videoStreamInfo = VideoStreamInfo.translateFromRPC(response.videoStreamInfo)
+                    observer.onNext(videoStreamInfo)
+                }
+            } catch {
+                observer.onError("Failed to subscribe to video stream info. \(error)")
+            }
+            
+            return Disposables.create()
+        }
+    }
+
+    public func createCaptureInfoObservable() -> Observable<CaptureInfo> {
+        return Observable.create { observer in
+            let captureInfoRequest = Dronecore_Rpc_Camera_SubscribeCaptureInfoRequest()
+            
+            do {
+                let call = try self.service.subscribecaptureinfo(captureInfoRequest, completion: nil)
+                while let response = try? call.receive() {
+                    let captureInfo = CaptureInfo.translateFromRPC(response.captureInfo)
+                    observer.onNext(captureInfo)
+                }
+            } catch {
+                observer.onError("Faile to subscribe to capture info. \(error)")
+            }
+            
+            return Disposables.create()
+        }
+    }
+    
+    public func getPossibleSettings() -> Single<[Setting]> {
```

Comment:
This one is now out of date (sorry about that, the [proto file](https://github.com/dronecore/DroneCore-Proto/blob/master/camera/camera.proto) should not change anymore).

The issue [here](https://github.com/dronecore/DroneCore-Swift/issues/5) is up-to-date with the proto file, BTW.